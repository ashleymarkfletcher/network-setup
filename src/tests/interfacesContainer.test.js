import React from 'react';
import renderer from 'react-test-renderer';
import InterfacesContainer from '../components/interfacesContainer';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

test('Interface renders', () => {

  const mockConfigs = [
    {
      errors: {},
      gateway: "192.168.1.1",
      id: "_8h7ez2224",
      ip: "192.168.1.10",
      primaryDNS: "8.8.8.8",
      secondaryDNS: "8.8.4.4",
      subnet: "255.255.255.0"
    },
    {
      errors: {},
      gateway: "192.168.21.1",
      id: "_8h7ez2244",
      ip: "192.168.21.10",
      primaryDNS: "8.8.8.8",
      secondaryDNS: "8.8.4.4",
      subnet: "255.255.255.0"
    },
    {
      errors: {},
      gateway: "192.168.40.1",
      id: "_8h7ez2232",
      ip: "192.168.40.12",
      primaryDNS: "8.8.8.8",
      secondaryDNS: "8.8.4.4",
      subnet: "255.255.255.0"
    }
  ]

  const configureInterface = () => {}

  const deleteConfig = () => {}

  const save = () => {}

  const dhcp = () => {}

  const tree = shallow(
      <InterfacesContainer
        configureInterface={configureInterface}
        save={save}
        configs={mockConfigs}
        deleteConfig={deleteConfig}
        dhcp={dhcp}
      >
      </InterfacesContainer>
  )

  expect(shallowToJson(tree)).toMatchSnapshot();
})
