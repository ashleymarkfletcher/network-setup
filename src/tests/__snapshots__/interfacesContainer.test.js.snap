// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interface renders 1`] = `
<div
  className="interface-container"
>
  <RaisedButton
    backgroundColor="#FFC107"
    disabled={false}
    fullWidth={false}
    icon={
      <ContentAddBox
        color="#555555"
      />
    }
    label="New Config"
    labelPosition="after"
    onClick={[Function]}
    primary={false}
    secondary={false}
    style={
      Object {
        "marginLeft": 10,
      }
    }
  />
  <RaisedButton
    backgroundColor="#607D8B"
    disabled={false}
    fullWidth={false}
    label="DHCP"
    labelPosition="after"
    onClick={[Function]}
    primary={false}
    secondary={false}
    style={
      Object {
        "marginLeft": 10,
      }
    }
  />
  <Interface
    config={
      Object {
        "errors": Object {},
        "gateway": "192.168.1.1",
        "id": "_8h7ez2224",
        "ip": "192.168.1.10",
        "primaryDNS": "8.8.8.8",
        "secondaryDNS": "8.8.4.4",
        "subnet": "255.255.255.0",
      }
    }
    configureInterface={[Function]}
    deleteConfig={[Function]}
    key="_8h7ez2224"
    save={[Function]}
  />
  <Interface
    config={
      Object {
        "errors": Object {},
        "gateway": "192.168.21.1",
        "id": "_8h7ez2244",
        "ip": "192.168.21.10",
        "primaryDNS": "8.8.8.8",
        "secondaryDNS": "8.8.4.4",
        "subnet": "255.255.255.0",
      }
    }
    configureInterface={[Function]}
    deleteConfig={[Function]}
    key="_8h7ez2244"
    save={[Function]}
  />
  <Interface
    config={
      Object {
        "errors": Object {},
        "gateway": "192.168.40.1",
        "id": "_8h7ez2232",
        "ip": "192.168.40.12",
        "primaryDNS": "8.8.8.8",
        "secondaryDNS": "8.8.4.4",
        "subnet": "255.255.255.0",
      }
    }
    configureInterface={[Function]}
    deleteConfig={[Function]}
    key="_8h7ez2232"
    save={[Function]}
  />
</div>
`;
